src/Microsoft.OpenApi/Extensions/OpenApiTypeMapper.cs

```diff
@@ -0,0 +1,85 @@
+
+using System;
+using System.Collections.Generic;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.OpenApi.Extensions
+{
+    /// <summary>
+    /// Extension methods for <see cref="Type"/>.
+    /// </summary>
+    public static class OpenApiTypeMapper
+    {
+        private static readonly Dictionary<Type, Func<OpenApiSchema>> _simpleTypeToOpenApiSchema = new()
+        {
+            [typeof(bool)] = () => new OpenApiSchema { Type = "boolean" },
+            [typeof(byte)] = () => new OpenApiSchema { Type = "string", Format = "byte" },
+            [typeof(int)] = () => new OpenApiSchema { Type = "integer", Format = "int32" },
+            [typeof(uint)] = () => new OpenApiSchema { Type = "integer", Format = "int32" },
+            [typeof(long)] = () => new OpenApiSchema { Type = "integer", Format = "int64" },
+            [typeof(ulong)] = () => new OpenApiSchema { Type = "integer", Format = "int64" },
+            [typeof(float)] = () => new OpenApiSchema { Type = "number", Format = "float" },
+            [typeof(double)] = () => new OpenApiSchema { Type = "number", Format = "double" },
+            [typeof(decimal)] = () => new OpenApiSchema { Type = "number", Format = "double" },
+            [typeof(DateTime)] = () => new OpenApiSchema { Type = "string", Format = "date-time" },
+            [typeof(DateTimeOffset)] = () => new OpenApiSchema { Type = "string", Format = "date-time" },
+            [typeof(Guid)] = () => new OpenApiSchema { Type = "string", Format = "uuid" },
+            [typeof(char)] = () => new OpenApiSchema { Type = "string" },
+            
+            // Nullable types
+            [typeof(bool?)] = () => new OpenApiSchema { Type = "boolean", Nullable = true },
+            [typeof(byte?)] = () => new OpenApiSchema { Type = "string", Format = "byte", Nullable = true },
+            [typeof(int?)] = () => new OpenApiSchema { Type = "integer", Format = "int32", Nullable = true },
+            [typeof(uint?)] = () => new OpenApiSchema { Type = "integer", Format = "int32", Nullable = true },
+            [typeof(long?)] = () => new OpenApiSchema { Type = "integer", Format = "int64", Nullable = true },
+            [typeof(ulong?)] = () => new OpenApiSchema { Type = "integer", Format = "int64", Nullable = true },
+            [typeof(float?)] = () => new OpenApiSchema { Type = "number", Format = "float", Nullable = true },
+            [typeof(double?)] = () => new OpenApiSchema { Type = "number", Format = "double", Nullable = true },
+            [typeof(decimal?)] = () => new OpenApiSchema { Type = "number", Format = "double", Nullable = true },
+            [typeof(DateTime?)] = () => new OpenApiSchema { Type = "string", Format = "date-time", Nullable = true },
+            [typeof(DateTimeOffset?)] = () => new OpenApiSchema { Type = "string", Format = "date-time", Nullable = true },
+            [typeof(Guid?)] = () => new OpenApiSchema { Type = "string", Format = "uuid", Nullable = true },
```

Comment:
Is `uuid` a valid format?

Reply:
Yep
https://spec.openapis.org/registry/format/ 
https://json-schema.org/draft/2020-12/json-schema-validation.html#name-vocabularies-for-semantic-c
https://json-schema.org/draft/2020-12/json-schema-validation.html#name-defined-formats